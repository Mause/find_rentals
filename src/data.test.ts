import { VercelRequest, VercelResponse } from '@vercel/node';
import endpoint from '../api/data';
import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
import dotenv from 'dotenv';
import { Polly } from '@pollyjs/core';
import XHRAdapter from '@pollyjs/adapter-xhr';
import RESTPersister from '@pollyjs/persister-rest';

Polly.register(XHRAdapter);
Polly.register(RESTPersister);

dotenv.config();

let polly: Polly;

beforeEach(() => {
  polly = new Polly('Data Test', {
    adapters: ['xhr'],
    persister: 'rest',
  });
  polly.record();
});
afterEach(async () => {
  await polly.stop();
});

describe('data', () => {
  it('hello', async () => {
    const json = jest.fn(
      (body: any): VercelResponse => {
        return res;
      }
    );
    const res = ({ json } as unknown) as VercelResponse;

    jest.setTimeout(500000);
    await endpoint(({} as unknown) as VercelRequest, res);

    expect(json.mock.calls[0][0]).toMatchSnapshot();
  });
});
