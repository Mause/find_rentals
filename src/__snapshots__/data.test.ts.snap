// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data hello 1`] = `
Object {
  "rows": Array [
    Object {
      "Address": "367 Crawford Rd, INGLEWOOD",
      "Applied?": "Yes- 4/4/21 - 2Apply",
      "Beds": 3,
      "Concerns": "Noisy ",
      "Good things": "",
      "Interested": Array [
        "CM",
        "EM",
        "EW",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-inglewood-431575634",
      "Price": "$570",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 22/4",
      "Viewed?": "Yes - EW, EM, CM",
      "applicationStatus": "unknown",
      "system": "TWO_APPLY",
    },
    Object {
      "Address": "122 Stuart Street, INGLEWOOD",
      "Applied?": "Yes- 4/4/21 - 2Apply",
      "Beds": 4,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "",
      "Price": "$650",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful - 12/04/2021",
      "Viewed?": "Yes- Emery",
      "applicationStatus": "unknown",
      "system": "TWO_APPLY",
    },
    Object {
      "Address": "55A Gloucester Street, Victoria Park",
      "Applied?": "Yes - 13/04/21 - emailed by Elliana",
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-victoria+park-423683714",
      "Price": "$675",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful - 18/04/2021",
      "Viewed?": "Yes - Emery & Elliana",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "24 Pennant St, NORTH PERTH",
      "Applied?": "Yes- 4/4/21 submitted by Emery",
      "Beds": 4,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
        "CHW",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-north+perth-431548518",
      "Price": "$850",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful - 12/04/2021",
      "Viewed?": "Yes - Emery",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "64 Sydney St, North Perth, WA 6006",
      "Applied?": "Courtney not interested",
      "Beds": 3,
      "Concerns": "EW - I would want to request to have a locking back screen door installed, that way we can let in a breeze etc (it's just a wooden door at the moment)",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-north+perth-431592686",
      "Price": "$495",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Yes - Emery",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "46 First Avenue, Mount Lawley",
      "Applied?": "Yes - 16/4 - 2Apply",
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-mount+lawley-428739594",
      "Price": "$650",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 21/4",
      "Viewed?": "Yes - Elliana",
      "applicationStatus": "unknown",
      "system": "TWO_APPLY",
    },
    Object {
      "Address": "20A Mabel Street, North Perth",
      "Applied?": "Yes - 11/4/21 submitted by Emery",
      "Beds": 3,
      "Concerns": "Small secondary rooms",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-north+perth-431627734",
      "Price": "$625",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 16/4",
      "Viewed?": "Yes - Emery & Elliana",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "244 Gloucester St VIC PARK",
      "Applied?": "Yes - 11/4/21 submitted by Courtney ",
      "Beds": 4,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-victoria+park-431612826",
      "Price": "$600",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 16/4",
      "Viewed?": "Yes - Courtney",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "3/62 Golf St, YOKINE",
      "Applied?": "too late",
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-yokine-431645454",
      "Price": "$500",
      "RealStatus": "Gone before we applied",
      "Status": undefined,
      "Viewed?": "EW email sent",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "30 Hartwell St, COOLBINIA",
      "Applied?": "too late",
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-coolbinia-135521298",
      "Price": "$650",
      "RealStatus": "Gone before we applied",
      "Status": undefined,
      "Viewed?": "Yes -Emery ",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "45 Armagh St VIC PARK",
      "Applied?": "too late",
      "Beds": 4,
      "Concerns": "Pricey",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-victoria+park-431642722",
      "Price": "$680",
      "RealStatus": "Gone before we applied",
      "Status": undefined,
      "Viewed?": "Didn't make it to sat 17th",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "46 Axon St, SUBIACO ",
      "Applied?": "Elliana not interested",
      "Beds": 3,
      "Concerns": "Ugly tiles ",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-subiaco-431655258",
      "Price": "$500",
      "RealStatus": "Gone before we applied",
      "Status": undefined,
      "Viewed?": "Canceled ",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "3/8 Withnell St EAST VIC PARK",
      "Applied?": "Yes - 24/04/2021 - 2Apply",
      "Beds": 3,
      "Concerns": "",
      "Good things": "CHEAP",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-east+victoria+park-431655690",
      "Price": "$450",
      "RealStatus": "Application pending",
      "Status": "Pending",
      "Viewed?": "Didn't make it",
      "applicationStatus": "unknown",
      "system": "TWO_APPLY",
    },
    Object {
      "Address": "142 Carnarvon St EAST VIC PARK",
      "Applied?": "Yes - 15/4 submitted by Emery",
      "Beds": 3,
      "Concerns": "laundry is in bathroom",
      "Good things": "comes with washing machine, good price",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-east+victoria+park-431591598",
      "Price": "$560",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 21/4",
      "Viewed?": "Yes - Emery",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "115 Westminster St EAST VIC PARK",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-east+victoria+park-431660726",
      "Price": "$525",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Fri 30th April 5.15pm",
    },
    Object {
      "Address": "38A Armada St BAYSWATER ",
      "Applied?": "too late",
      "Beds": 3,
      "Concerns": "A bit further north then we r hoping for",
      "Good things": "A huge theatre room with it looks like a projector!?",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.domain.com.au/a-38-armada-street-bayswater-wa-6053-15047847?utm_source=www.domain.com.au&utm_medium=sharelisting",
      "Price": "$600",
      "RealStatus": "Gone before we applied",
      "Status": undefined,
      "Viewed?": "EW email sent ",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "22C Stamford St LEEDERVILLE ",
      "Applied?": "Not interested",
      "Beds": 3,
      "Concerns": "Pricey, might be noisy?",
      "Good things": "",
      "Interested": Array [],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-leederville-431493722",
      "Price": "$700",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Canceled",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "100 Blencowe St WEST LEEDERVILLE ",
      "Applied?": "Not interested after inspection.",
      "Beds": 3,
      "Concerns": "Way too run down (only obvious at inspection)",
      "Good things": "",
      "Interested": Array [],
      "Link": "https://www.domain.com.au/100-blencowe-street-west-leederville-wa-6007-15027064?utm_source=www.domain.com.au&utm_medium=sharelisting",
      "Price": "$595",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Yes - Emery",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "12 York Street, INGLEWOOD",
      "Applied?": "Yes - 24/04/2021 - 1Form",
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-inglewood-417769938",
      "Price": "$620",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 27/4",
      "Viewed?": "Yes - Emery Code: RADI6052",
      "applicationStatus": "unknown",
      "system": "ONE_FORM",
    },
    Object {
      "Address": "63 Federation St, MOUNT HAWTHORN ",
      "Applied?": "Not interested",
      "Beds": 3,
      "Concerns": "6 month lease only",
      "Good things": "",
      "Interested": Array [],
      "Link": "https://www.realestate.com.au/property-house-wa-mount+hawthorn-431556606",
      "Price": "$625",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "EW email sent",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "26 Swansea Street, East Victoria Park, WA 6101",
      "Applied?": "Not interested",
      "Beds": 4,
      "Concerns": "six month max lease.",
      "Good things": "",
      "Interested": Array [
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-east+victoria+park-431660014",
      "Price": "$600",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Yes - EM",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "12 Abbotsford st WEST LEEDERVILLE",
      "Applied?": "too late",
      "Beds": 3,
      "Concerns": "small, under application",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-west+leederville-431700590",
      "Price": "$520",
      "RealStatus": "Gone before we applied",
      "Status": undefined,
      "Viewed?": "EW email sent",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "16A Bourke St, NORTH PERTH",
      "Applied?": "not interested",
      "Beds": 3,
      "Concerns": "small 3rd bedroom, 120m from power substation",
      "Good things": "",
      "Interested": Array [
        "EW",
        "CM",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-north+perth-431699822",
      "Price": "$510",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Yes - EW Wed 28th 2pm",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "16 Camelia Street, NORTH PERTH",
      "Applied?": "yes - 27/4 - 1Form",
      "Beds": 4,
      "Concerns": "",
      "Good things": "pool!",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
        "CHW",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-north+perth-431719110",
      "Price": "$785",
      "RealStatus": "Application pending",
      "Status": "Pending",
      "Viewed?": "yes - EW Tues 27th 3.50pm CODE: TPE6008",
      "applicationStatus": "unknown",
      "system": "ONE_FORM",
    },
    Object {
      "Address": "2/103 Swansea Street, EAST VIC PARK",
      "Applied?": "Yes - 25/04/21 -Email by EW",
      "Beds": 3,
      "Concerns": "Villa",
      "Good things": "",
      "Interested": Array [
        "EM",
        "CM",
        "EW",
      ],
      "Link": "https://www.realestate.com.au/property-villa-wa-east+victoria+park-431724322",
      "Price": "$450",
      "RealStatus": "Application pending",
      "Status": "Pending",
      "Viewed?": "EW email sent",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "16B Dane St, EAST VIC PARK",
      "Applied?": "not interested",
      "Beds": 3,
      "Concerns": " bed 2+3 are small",
      "Good things": "",
      "Interested": Array [
        "EM",
        "CM",
        "EW",
      ],
      "Link": "https://www.realestate.com.au/property-villa-wa-east+victoria+park-431724318",
      "Price": "$495",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Yes - Wed 28th April 4.45pm EM",
      "system": "UNKNOWN",
    },
    Object {
      "Address": "17 Coode St, MOUNT LAWLEY",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "no photos online?",
      "Good things": "",
      "Interested": Array [],
      "Link": "https://www.realestate.com.au/property-house-wa-mount+lawley-431728402",
      "Price": "$620",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "EW email sent",
    },
    Object {
      "Address": "197 Eighth Avenue, INGLEWOOD",
      "Applied?": "Yes - 24/04/2021 - 1Form offered $570",
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-inglewood-431724986",
      "Price": "$550",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 27/4",
      "Viewed?": "Yes - Emery Code: RADI6052",
      "applicationStatus": "unknown",
      "system": "ONE_FORM",
    },
    Object {
      "Address": "156 Sussex St, EAST VIC PARK",
      "Applied?": undefined,
      "Beds": 4,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-east+victoria+park-431724270",
      "Price": "$600",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Sat 1St May 9.30am",
    },
    Object {
      "Address": "4/202 Bishopsgate Street, Carlisle, WA, 6101",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "townhouse, bedrooms too small",
      "Good things": "",
      "Interested": Array [
        "EM",
      ],
      "Link": "https://m.realestate.com.au/property-townhouse-wa-carlisle-431682458",
      "Price": "$520",
      "RealStatus": "Not interested",
      "Status": undefined,
      "Viewed?": "Yes CM- Tues 27th 3.15pm",
    },
    Object {
      "Address": "3a View St, MAYLANDS",
      "Applied?": undefined,
      "Beds": 5,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-maylands-431730366",
      "Price": "$650",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Sat 1st May 10.20am EW",
    },
    Object {
      "Address": "26 Gresham St, VIC PARK",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-victoria+park-431729218",
      "Price": "$500",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Sat 1st May 10am EM",
    },
    Object {
      "Address": "11B Little Russel St, NORTH PERTH",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "Townhouse",
      "Good things": "",
      "Interested": Array [
        "EM",
        "EW",
      ],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-north+perth-431652478",
      "Price": "$480",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Sat 1st Mat 1.50pm  EW",
    },
    Object {
      "Address": "2/28 Hill View Terrace, ST JAMES",
      "Applied?": "CM to complete 1Form",
      "Beds": 5,
      "Concerns": "Tiles",
      "Good things": "6 beds!?!",
      "Interested": Array [
        "EM",
        "EW",
        "CM",
      ],
      "Link": "https://www.domain.com.au/2-28-hill-view-terrace-st-james-wa-6102-15075698?utm_source=www.domain.com.au&utm_medium=sharelisting",
      "Price": "$550",
      "RealStatus": "Urgent attention",
      "Status": undefined,
      "Viewed?": "yes -Wed 28th 4.20pm EM",
      "applicationStatus": "unknown",
      "system": "ONE_FORM",
    },
    Object {
      "Address": "20/431 Baggot Rd, SUBIACO",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "Townhouse, staircase not safe/accessible",
      "Good things": "",
      "Interested": Array [
        "EM",
      ],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-subiaco-431735810",
      "Price": "$550",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Sat 1st May 11.30am",
    },
    Object {
      "Address": "4/10 Queens crescent, MOUNT LAWLEY",
      "Applied?": "Yes - 27/4 offered $500 - 1Form",
      "Beds": 3,
      "Concerns": "Aircon only in kitchen/dining ",
      "Good things": "",
      "Interested": Array [
        "EW",
        "EM",
        "CM",
      ],
      "Link": "https://www.realestate.com.au/property-townhouse-wa-mount+lawley-431743674",
      "Price": "$480",
      "RealStatus": "Unsuccessful application",
      "Status": "Unsuccessful 28/4",
      "Viewed?": "Yes - EW Tues 27th 10.30am",
      "applicationStatus": "unknown",
      "system": "ONE_FORM",
    },
    Object {
      "Address": "72 Swansea St, EAST VIC PARK",
      "Applied?": undefined,
      "Beds": 3,
      "Concerns": "",
      "Good things": "",
      "Interested": Array [
        "EW",
      ],
      "Link": "https://www.realestate.com.au/property-house-wa-east+victoria+park-431762002",
      "Price": "$580",
      "RealStatus": "Need to decide if interested",
      "Status": undefined,
      "Viewed?": "Sat 1st May 9.15am",
    },
  ],
  "statusMapping": Object {
    "Application pending": "{\\"color\\":\\"rgba(235, 210, 221, 100)\\",\\"strikethrough\\":false}",
    "Concern": "{\\"color\\":\\"rgba(235, 154, 154, 100)\\",\\"strikethrough\\":false}",
    "Excellent": "{\\"color\\":\\"rgba(0, 256, 0, 100)\\",\\"strikethrough\\":false}",
    "Gone before we applied": "{\\"color\\":\\"rgba(102, 102, 102, 100)\\",\\"strikethrough\\":true}",
    "Good": "{\\"color\\":\\"rgba(183, 216, 169, 100)\\",\\"strikethrough\\":false}",
    "Need to decide if interested": "{\\"color\\":\\"rgba(111, 169, 221, 100)\\",\\"strikethrough\\":false}",
    "Needs a fourth person": "{\\"color\\":\\"rgba(0, 256, 256, 100)\\",\\"strikethrough\\":false}",
    "Not interested": "{\\"strikethrough\\":true}",
    "Price has dropped": "{\\"color\\":\\"rgba(181, 168, 215, 100)\\",\\"strikethrough\\":false}",
    "Problem": "{\\"color\\":\\"rgba(225, 102, 102, 100)\\",\\"strikethrough\\":false}",
    "Something to do": "{\\"color\\":\\"rgba(256, 256, 0, 100)\\",\\"strikethrough\\":false}",
    "Unsuccessful application": "{\\"color\\":\\"rgba(154, 154, 154, 100)\\",\\"strikethrough\\":false}",
    "Urgent attention": "{\\"color\\":\\"rgba(256, 154, 0, 100)\\",\\"strikethrough\\":false}",
  },
  "title": "Housing",
}
`;
